docs/supported-services/stepfunctions.rst

```diff
@@ -0,0 +1,106 @@
+.. _stepfunctions:
+
+Step Functions
+==============
+This document contains information about the Step Functions service supported in Handel. This Handel service provisions Step Functions state machine resources to provide an application workflow.
+
+Service Limitations
+-------------------
+No Activities
+~~~~~~~~~~~~~
+This service does not yet support Step Functions activity resources. Task resources are limited to Lambda functions.
+
+Parameters
+----------
+
+.. list-table::
+   :header-rows: 1
+
+   * - Parameter
+     - Type
+     - Required
+     - Default
+     - Description
+   * - type
+     - string
+     - Yes
+     - 
+     - This must always be *stepfunctions* for this service type.
+   * - definition
+     - :ref:`stepfunctions-definition`
+     - Yes
+     -
+     - Path to file containing state machine definition.
+
+.. _stepfunctions-definition:
+
+State Machine Definition
+~~~~~~~~~~~~~~~~~~~~~~~~
+For the most part, the definition file you provide in the *definition* section is in `Amazon States Language <https://docs.aws.amazon.com/step-functions/latest/dg/concepts-amazon-states-language.html>`_. Instead of providing an ARN in the 'Resource' field of a state, however, one should give the service name from the Handel file. For convenience, Handel supports both JSON and YAML formats for the definition file, where pure States Language is based on JSON alone.
```

Comment:
Can we make the last sentence starting with "For convenience" a Note box so that it pops out a bit more?

Also, I might recommend taking the example state_machine.yml file in the Example Handel File section and put it in this section instead, since this section is helping you figure out what needs to be provided for the *definition* parameter. Then you could put a comment in the example Handel file on the *definition* line that refers them to this section if they have questions about what to put in that file.